dask_kubernetes/utils.py

```diff
@@ -53,8 +53,10 @@ async def get_external_address_for_scheduler_service(
         lb = service.status.load_balancer.ingress[0]
         host = lb.hostname or lb.ip
     elif service.spec.type == "NodePort":
-        nodes = await core_api.list_node()
-        host = nodes.items[0].status.addresses[0].address
+        host = nodeport_host
+        if host is None:
+            nodes = await core_api.list_node()
+            host = nodes.items[0].status.addresses[0].address
```

Comment:
Small nit to improve readibility.

```suggestion
        if nodeport_host is not None:
            host = nodeport_host
        else:
            nodes = await core_api.list_node()
            host = nodes.items[0].status.addresses[0].address
```